src/Axe/Controllers/CompilerController.cs

```diff
@@ -0,0 +1,125 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Axe.Models;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Emit;
+using System.IO;
+using System.Reflection;
+using System.Collections;
+using System.Runtime.Loader;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Axe.Controllers
+{
+    public class CompilerController : ControllerExt
+    {
+        public CompilerController(UserManager<ApplicationUser> userManager, AxeDbContext context)
+            : base(userManager, context)
+        {
+        }
+
+        [HttpGet]
+        public IActionResult Index()
+        {
+            List<CodeBlock> codeBlocks = this.context.CodeBlock.ToList();
+            return View(codeBlocks);
+        }
+
+        [HttpGet]
+        public IActionResult Solve(int id)
+        {
+            try
+            {
+                CodeBlock codeBlock = context.CodeBlock.First(c => c.Id == id);
+                codeBlock.SourceCode = CSharpSyntaxTree.ParseText(codeBlock.SourceCode).GetRoot().NormalizeWhitespace().ToFullString(); // Source code formatting
+                return View(codeBlock);
+            }
+            catch (InvalidOperationException)
+            {
+                return NotFound();
+            }
+        }
+
+        [HttpPost]
+        public IActionResult Solve(CodeBlock codeBlock)
```

Comment:
Too much code in this method. Move all compilation and verification logic to separate classes (business layer) so that they can be unit-testable.